src/main/java/org/truffleruby/extra/TrufflePosixNodes.java

```diff
@@ -56,32 +54,4 @@ public DynamicObject memset(DynamicObject pointer, int c, long length) {
 
     }
 
-    @CoreMethod(names = "major", isModuleFunction = true, required = 1)
-    public abstract static class MajorNode extends CoreMethodArrayArgumentsNode {
-
-        @Specialization
-        public int major(long dev) {
-            if (Platform.getPlatform().getOS() == OS.SOLARIS) {
-                return (int) (dev >> 32); // Solaris has major number in the upper 32 bits.
```

Comment:
Why don't we need to do something like this any longer? Is it that `dev_t` is properly typed now? Or does NFI just handle endianness better?

Reply:
The commit (ed3057a9b4add791ebf5d8b5cb25ea73acacbe9c) should clarify.
We can just use the system macros for it, no need to hardcode values.

Reply:
Thanks. I misread and thought the return value needed to be treated.